### DISCUSSION TITLE
How to Extend Canvas with Streaked Edges

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#7712] [Help] [answered]
### ImageMagick version

7.1.0-62-Q16

### Operating system, version and so on

Windows 10

### Description

Hi everyone,

I’m trying to use the -extent option in ImageMagick to double the canvas size of an existing image, while keeping the original image intact. The goal is to extend the canvas vertically (top and bottom) and fill the new space with a streaked effect from the image’s edges, using the -virtual-pixel edge setting.

Here's the command I've been using:

magick in.jpg -virtual-pixel edge -gravity center -extent 100%%x200%% out.png

However, it seems like the -virtual-pixel edge setting isn’t affecting the -extent command the way I expected. I was hoping the new canvas space would be filled with streaked edges, but that doesn’t seem to be happening.

I’m sure there are multiple ways to accomplish this, but I’d like to keep things as straightforward as possible. Before diving deeper, I wanted to check if I’m missing something obvious. Any suggestions or insights would be greatly appreciated!

Thanks in advance!

**Comment by fmw42** [comment]
You can do that with viewport distort as follows --- Unix syntax.  (Sorry, I do not know how to specify a variable in Windows syntax. A Windows user can convert this for you. Or just replace $pad with a fixed number and possible need to double the % to %%)

![barn](https://github.com/user-attachments/assets/caa0c78a-80af-4e05-ba2c-dbefe802f3d3)

```
pad=100
magick barn.jpg  -set option:distort:viewport "%[fx:w+2*$pad]x%[fx:h+2*$pad]-%[fx:$pad]-%[fx:$pad]"  -filter point -virtual-pixel edge -distort SRT 0  +repage barn_viewport.jpg

```

![barn_viewport](https://github.com/user-attachments/assets/32f404f0-a52b-4a5b-a9e1-517ab172c963)


Note that -virtual-pixel edge is the default and could be removed. But you can use any of the other methods such as any of the tile options or mirror or random to create other edge effects.

**Reply by ghost** [reply, marked as answer]
Thanks Fred! Yes, that works perfectly after I made the syntax changes for windows.

**Reply by fmw42** [reply]
Post the Windows version to help anyone else on Windows.

**Comment by ghost** [comment]
Windows Version:

magick in.jpg ^
  -set option:distort:viewport %%[fx:w]x%%[fx:h*2] ^
  -virtual-pixel Edge -gravity Center ^
  -distort SRT "0,0 1,1 0 0,%%[fx:h/2]" ^
  out.jpg

**Reply by fmw42** [reply]
That is not a straight conversion.  What effect are you trying to achieve?  Could you show and input and output example?

**Reply by ghost** [reply]
I'm a digital artist and experiment with abstract designs and patterns. I have no set goal and I'm constantly experimenting and fiddling.
